snapm/manager/_boot.py

```diff
@@ -53,6 +58,123 @@
+        It populates the self.entries list.
+        """
+        try:
+            with open(self.path, 'r', encoding="utf8") as f:
+                for line in f:
+                    # 1. Strip leading/trailing whitespace.
+                    line = line.strip()
+
+                    # 2. Ignore empty lines and comments.
+                    if not line or line.startswith('#'):
+                        continue
+
+                    # 3. Split the line into parts.
+                    parts = line.split()
+
+                    # 4. An fstab entry must have 6 fields.
+                    if len(parts) != 6:
+                        # You might want to log a warning here in a real application.
+                        continue
+
+                    # 5. Convert freq and passno to integers for proper typing.
+                    try:
+                        parts[4] = int(parts[4]) # freq
+                        parts[5] = int(parts[5]) # passno
+                    except KeyError:
+                        continue
+
+                    # 6. Create a named tuple and append it to our list.
+                    entry = self.FstabEntry(*parts)
+                    self.entries.append(entry)
+
+        except FileNotFoundError:
+            _log_error(f"Error: The file '{self.path}' was not found.")
+            raise SnapmNotFoundError(f"Fstab file not found: {self.path}")
+        except IOError as e:
+            _log_error(f"Error: Could not read the file '{self.path}': {e}")
+            raise SnapmSystemError(f"Error reading fstab file: {self.path}") from e
+
+    def __iter__(self):
+        """
+        Allows iteration over the fstab entries.
+
+        Yields:
+            A 6-tuple for each entry in the fstab file:
+            (what, where, fstype, options, freq, passno)
+        """
+        for entry in self.entries:
+            yield entry
+
+    def lookup(self, key, value):
+        """
+        Finds and generates all entries matching a specific key-value pair.
+
+        Args:
+            key (str): The field to search by. Must be one of 'what', 'where',
+                       'fstype', 'options', 'freq', or 'passno'.
+            value: The value to match for the given key.
+
+        Yields:
+            A 6-tuple for each matching fstab entry.
+
+        Raises:
+            ValueError: If the provided key is not a valid fstab field name.
+        """
+        if key not in self.FstabEntry._fields:
+            raise KeyError(
+                f"Invalid lookup key: '{key}'. "
+                f"Valid keys are: {self.FstabEntry._fields}"
+            )
+
+        for entry in self.entries:
+            # getattr() allows us to get an attribute by its string name.
+            if getattr(entry, key) == value:
+                yield entry
+
+    def __repr__(self):
+        """Provides a developer-friendly representation of the object."""
+        return f"Fstab(path='{self.path}', entries={len(self.entries)})"
```

Comment:
Generally the result of calling `repr()` on an object of type Foo should return a string in the format of the `Foo.__init__(a, b, c)`, arguments, like `Foo(1, 2, 3)`, so we should probably just include `path=` here, and drop the `entries=NUM` from the repr return value.